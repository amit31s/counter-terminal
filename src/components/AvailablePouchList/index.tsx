import { PouchAcceptanceDetails } from "@ct/api/generator";
import { useGetPouchForAcceptance } from "@ct/features/PouchAcceptanceFeature/useGetPouchForAcceptance";
import { priceToRender } from "@ct/utils";
import { useMemo, useState } from "react";
import { FlatList, Text, TouchableHighlight, View } from "react-native";
import { stringConstants } from "../../constants";
import colorConstants from "../../constants/ColorConstants";
import styles from "./styles";

interface AvailablePouchListProps {
  availablePouches: PouchAcceptanceDetails[];
  parentCallback: (pouch: PouchAcceptanceDetails) => void;
  pouchValueAddition?: string;
}

const AvailablePouchList = ({ availablePouches, parentCallback }: AvailablePouchListProps) => {
  const { validatedData } = useGetPouchForAcceptance();

  availablePouches = useMemo(() => {
    const validatedBarcodes = validatedData.map((validatedPouch) => validatedPouch.pouchID);
    if (!availablePouches || !Array.isArray(availablePouches)) {
      return [];
    }
    return availablePouches.filter(
      (availablePouch) => !validatedBarcodes.includes(availablePouch.pouchID),
    );
  }, [availablePouches, validatedData]);

  const [selectedId, setSelectedId] = useState<string | null>(null);
  const touchProps = {
    activeOpacity: 1,
    underlayColor: colorConstants.disabledTextColour,
  };

  const handleClick = (item: PouchAcceptanceDetails) => {
    if (!item.pouchID) {
      return;
    }
    setSelectedId(item.pouchID);
    parentCallback(item);
  };

  const renderItem = ({ item, index }: { item: PouchAcceptanceDetails; index: number }) => {
    return (
      <TouchableHighlight
        {...touchProps}
        testID={`post-row-${index}`}
        onPress={() => handleClick(item)}
        style={[styles.boxView, styles.tableRow]}
      >
        <View style={styles.flexRow}>
          <View style={styles.width50}>
            <Text style={styles.primaryText}>{item.pouchID}</Text>
          </View>
          <View style={[styles.width50, styles.pouchType]}>
            <Text style={[styles.primaryText, styles.textAlignCenter, styles.pouchTypeValue]}>
              {item.pouchType}
            </Text>
          </View>
          <View style={styles.width50}>
            <Text style={[styles.primaryText, styles.textAlignRight]}>
              {priceToRender(Math.abs(item?.totalValue ?? 0))}
            </Text>
          </View>
        </View>
      </TouchableHighlight>
    );
  };

  const renderHeader = () => {
    return (
      <View style={[styles.tableRow, styles.boxView, styles.flexRow]}>
        <View style={styles.width50}>
          <Text style={styles.barcodeTxtText}>{stringConstants.Pouch.BarCode}</Text>
        </View>
        <View style={styles.width50}>
          <Text style={[styles.tableRowHeaderText, styles.textAlignCenter]}>Pouch Type</Text>
        </View>
        <View style={styles.width50}>
          <Text style={[styles.tableRowHeaderText, styles.textAlignRight]}>
            {stringConstants.Pouch.PouchValue}
          </Text>
        </View>
      </View>
    );
  };

  return (
    <View>
      {renderHeader()}
      <FlatList
        testID="Body"
        data={availablePouches}
        renderItem={renderItem}
        keyExtractor={(item) => item.pouchID || item.pouchID || ""}
        extraData={selectedId}
      />
    </View>
  );
};

export default AvailablePouchList;
