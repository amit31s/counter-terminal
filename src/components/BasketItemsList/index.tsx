export { default as BasketItemsList } from "./BasketItemsList";
