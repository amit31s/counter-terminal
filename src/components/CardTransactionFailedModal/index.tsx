export { default as CardTransactionFailedModal } from "./CardTransactionFailedModal";
