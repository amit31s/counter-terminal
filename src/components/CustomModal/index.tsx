export { default as CustomModal } from "./CustomModal";
export * from "./CustomModal";
