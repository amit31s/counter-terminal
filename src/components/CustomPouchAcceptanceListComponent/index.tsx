import { PouchAcceptanceDetails } from "@ct/api/generator";
import { StyledDeleteIcon } from "@ct/assets/icons";
import { priceToRender } from "@ct/utils";
import { FlatList, Pressable, Text, View } from "react-native";
import stringConstants from "../../constants/StringsConstants";
import styles from "./style";

interface CustomPouchAcceptanceListComponentProps {
  parentdata: PouchAcceptanceDetails[];
  showDelete?: boolean;
  onDeleteItem?: (barcode: string) => void;
}

interface ItemViewProps {
  index: number;
  item: PouchAcceptanceDetails;
}

const ListHeader = ({ showDelete }: { showDelete: boolean }) => {
  return (
    <View style={[styles.textContainer, styles.headerFooterStyle, styles.border]}>
      <Text style={[styles.primaryText, styles.textStyle, styles.textAlignLeft]}>
        {stringConstants.Pouch.BarCode}
      </Text>
      {showDelete ? (
        <Text style={[styles.primaryText, styles.textStyle, styles.textAlignLeft]}>
          {stringConstants.Pouch.PouchType}
        </Text>
      ) : (
        <Text style={[styles.primaryText, styles.textStyle]}>
          {stringConstants.Pouch.PouchType}
        </Text>
      )}
      <Text style={[styles.primaryText, styles.textStyle, styles.pouchValue]}>
        {stringConstants.Pouch.PouchValue}
      </Text>
      {showDelete ? (
        <Text style={[styles.primaryText, styles.textStyle]}>{stringConstants.Pouch.Branch}</Text>
      ) : null}
      {showDelete ? <Text style={[styles.primaryText, styles.actionStyle]}>{""}</Text> : null}
    </View>
  );
};

const EmptyListMessage = () => <Text style={styles.emptyListStyle}>No Data Found</Text>;

const CustomPouchAcceptanceListComponent = ({
  parentdata,
  onDeleteItem,
  showDelete = true,
}: CustomPouchAcceptanceListComponentProps) => {
  const itemView = ({ item, index }: ItemViewProps) => {
    return (
      <View style={[styles.textContainer, styles.border]} testID={`pouch-row-${index}`}>
        <Text style={[styles.primaryText, styles.itemStyle, styles.textAlignLeft]}>
          {item.pouchID}
        </Text>
        <Text style={[styles.primaryText, styles.itemStyle, styles.pouchTypeValue]}>
          {item.pouchType}
        </Text>
        <Text style={[styles.primaryText, styles.itemStyle, styles.pouchValue]}>
          {priceToRender(Math.abs(item?.totalValue ?? 0))}
        </Text>

        {showDelete ? (
          <>
            <Text style={[styles.primaryText, styles.itemStyle]}>{item.assignedBranchName}</Text>
            <Pressable
              onPress={() => {
                if (onDeleteItem && item.pouchID) {
                  onDeleteItem(item.pouchID);
                }
              }}
              style={styles.actionStyle}
            >
              <StyledDeleteIcon />
            </Pressable>
          </>
        ) : null}
      </View>
    );
  };

  return (
    <FlatList
      testID="testFlatList"
      contentContainerStyle={styles.contentContainerStyle}
      data={parentdata}
      keyExtractor={(item: PouchAcceptanceDetails, index: number) =>
        item.pouchID || index.toString()
      }
      ListHeaderComponent={<ListHeader showDelete={showDelete} />}
      renderItem={itemView}
      ListEmptyComponent={EmptyListMessage}
      stickyHeaderIndices={[0]}
    />
  );
};

export default CustomPouchAcceptanceListComponent;
