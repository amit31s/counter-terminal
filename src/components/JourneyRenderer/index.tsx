import { getJourneyStatus, useAppSelector } from "@ct/common";
import { JOURNEY_LOGS_FN } from "@ct/common/constants/JourneyLogs";
import useKeyboardAvoider from "@ct/common/hooks/useKeyboardAvoider";
import { JourneyInterruptContext } from "@ct/screens/HomeScreen/JourneyInterruptContext";
import { LOGGER_TYPE } from "@pol/frontend-logger-common";
import { logManager } from "@pol/frontend-logger-web";
import { JourneyEngine, ProviderProps } from "postoffice-spm-journey-engine";
import { useCallback, useContext, useMemo, useState } from "react";
import { Dimensions, ScrollView, Text, View } from "react-native";
import { v4 } from "uuid";
import { stringConstants } from "../../constants";
import { enablerConfig } from "./enablerConfig";
import styles from "./style";
import { ReturnTypes, useAppData } from "./useAppData";
import { useDeviceCallback } from "./useDeviceCallback";
import { useLogger } from "./useLogger";
// Keeping as 'any' for Journey onComplete payload data because JourneyData type from Journey engine do not enforce many keys used in basket items basically its any key value based also changes required in CT for using this interface is big and would require a seperate ticket.
export type JourneyData = any;
const journeyEngineLogger = logManager(LOGGER_TYPE.journeyEngineLogger);

interface IJourneyRenderer {
  onJourneyAbort?: () => void;
  journeyOnComplete: (data: JourneyData) => Promise<void>;
  barcodeNotRecognizedCb: () => void;
}

export const JourneyRenderer = ({
  journeyOnComplete,
  onJourneyAbort,
  barcodeNotRecognizedCb,
}: IJourneyRenderer) => {
  const [journeyReturns, setJourneyReturns] = useState<ReturnTypes>();
  const appData = useAppData({ journeyReturns });
  const complianceStatus = true;
  const { keyboardPadding } = useKeyboardAvoider();
  const { session } = useAppSelector((rootState) => rootState.auth);
  const windowHeight = Dimensions.get("window").height / 1.7;
  const { interruptionInputData } = useContext(JourneyInterruptContext);
  const deviceCallback = useDeviceCallback(appData, journeyOnComplete);
  const journeyLogger = useLogger();
  const { open } = useAppSelector(getJourneyStatus);

  const contentContainerStyle = useMemo(
    () => ({ paddingBottom: keyboardPadding }),
    [keyboardPadding],
  );

  const getAuthHeaders = useCallback(
    async () => ({
      Authorization: session?.idToken ?? "",
      "X-Correlation-ID": v4(),
    }),
    [session?.idToken],
  );

  const onComplete: ProviderProps["onComplete"] = useCallback(
    (data) => {
      if (data && data.returns) {
        setJourneyReturns(data.returns);
      }
      journeyOnComplete(data);
    },
    [journeyOnComplete],
  );

  const onAbort: ProviderProps["onAbort"] = useCallback(
    (data) => {
      if (data && data.returns) {
        setJourneyReturns(data.returns);
      }
      onJourneyAbort?.();
    },
    [onJourneyAbort],
  );

  const onError: NonNullable<ProviderProps["onError"]> = useCallback(
    (error) => {
      journeyEngineLogger.error({
        methodName: JOURNEY_LOGS_FN.journeyEngineProvider,
        error: error as unknown as Error,
        data: interruptionInputData,
      });
      if (interruptionInputData?.value) {
        barcodeNotRecognizedCb();
      }
    },
    [barcodeNotRecognizedCb, interruptionInputData],
  );

  if (!complianceStatus) {
    return (
      <View style={styles(open).complianceCont}>
        <Text style={styles(open).compWarning}>
          {stringConstants.messages.outstandingCompliance}
        </Text>
      </View>
    );
  }

  return (
    <ScrollView style={styles(open).journeyContainer} contentContainerStyle={contentContainerStyle}>
      <JourneyEngine.Provider
        enablerConfig={enablerConfig}
        authHeaders={getAuthHeaders}
        appData={appData}
        logger={journeyLogger}
        interruptionInputData={interruptionInputData}
        onDeviceTrigger={deviceCallback}
        onComplete={onComplete}
        onAbort={onAbort}
        onError={onError}
      >
        <JourneyEngine.Canvas canvasMinHeight={windowHeight} />
      </JourneyEngine.Provider>
    </ScrollView>
  );
};

export type IJourneyData = JourneyData;
