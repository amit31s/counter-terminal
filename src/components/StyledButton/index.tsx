export { default as StyledButton } from "./StyledButton";
export * from "./StyledButton";
