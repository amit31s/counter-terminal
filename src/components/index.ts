export * from "./AvailablePouchList";
export * from "./BasketItemsList";
export * from "./CardTransactionFailedModal";
export * from "./CustomModal";
export * from "./HorizontalRule";
export * from "./LoadingModal";
export * from "./PinPadModal";
export * from "./StyledButton";
export * from "./StyledInput";
export * from "./SubmitPouchModal";
