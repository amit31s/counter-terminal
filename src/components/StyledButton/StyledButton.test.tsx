import { render, setupUser } from "@ct/common/helpers/test-utils";
import { StyledButton, StyledButtonSize, StyledButtonType } from "@ct/components";

describe("Styled button component renders correctly", () => {
  it("renders something", () => {
    const customStyle = { width: "20%" };
    const { getByTestId } = render(
      <StyledButton testID="ButtonTestId" label="Test Button" styles={[customStyle]} />,
    );
    expect(getByTestId("ButtonTestId")).toBeTruthy();
  });

  it("renders disable button", () => {
    const { getByTestId } = render(
      <StyledButton testID="ButtonTestId" label="Test Disable Button" isDisabled />,
    );
    expect(getByTestId("ButtonTestId")).toBeTruthy();
  });

  it("calls it's callback", async () => {
    const user = setupUser();
    const onPress = jest.fn();
    const { getByTestId } = render(
      <StyledButton testID="ButtonTestId" label="Test Button" onPress={onPress} size="slim" />,
    );
    await user.click(getByTestId("ButtonTestId"));
    expect(onPress).toHaveBeenCalledTimes(1);
  });

  const buttonTypes: StyledButtonType[] = [
    "primary",
    "secondary",
    "tertiary",
    "invertedPrimary",
    "invertedSecondary",
  ];
  const buttonSizes: StyledButtonSize[] = ["default", "slim"];

  buttonTypes.forEach((type) => {
    buttonSizes.forEach((size) => {
      it(`renders the ${size} ${type} variant correctly`, async () => {
        const view = render(
          <StyledButton testID="ButtonTestId" label="Test Button" type={type} size={size} />,
        );
        expect(view.baseElement).toMatchSnapshot();
      });
    });
  });
});
