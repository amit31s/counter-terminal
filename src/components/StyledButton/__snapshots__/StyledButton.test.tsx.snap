// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled button component renders correctly renders the default invertedPrimary variant correctly 1`] = `
<body>
  <div>
    <div
      class="css-view-1dbjc4n r-flex-13awgt0"
    >
      <div
        aria-label="Test Button"
        class="css-cursor-18t94o4 css-view-1dbjc4n r-alignItems-1awozwy r-alignSelf-1kihuf0 r-backgroundColor-14lw9ot r-borderColor-11mg6pl r-borderRadius-z2wwpe r-borderStyle-1phboty r-borderWidth-d045u9 r-cursor-1loqt21 r-height-yq7r3z r-justifyContent-1777fci r-minHeight-j5ldn5 r-paddingHorizontal-2q4g5t r-shadow-f5u5bj r-touchAction-1otgn73"
        data-testid="ButtonTestId"
        role="button"
        tabindex="0"
      >
        <div
          class="css-text-901oao r-color-cq6wur r-fontFamily-v9d72p r-fontSize-1x35g6 r-fontWeight-b88u0q r-userSelect-lrvibr"
          dir="auto"
        >
          Test Button
        </div>
      </div>
    </div>
  </div>
  <div
    style="position: fixed; left: 0px; top: 0px; width: 0px; height: 0px; z-index: -1; overflow: hidden; visibility: hidden; transition-duration: 0.05s; transition-property: padding; transition-delay: 0s;"
  />
</body>
`;

exports[`Styled button component renders correctly renders the default invertedSecondary variant correctly 1`] = `
<body>
  <div>
    <div
      class="css-view-1dbjc4n r-flex-13awgt0"
    >
      <div
        aria-label="Test Button"
        class="css-cursor-18t94o4 css-view-1dbjc4n r-alignItems-1awozwy r-alignSelf-1kihuf0 r-backgroundColor-1niwhzg r-borderColor-11mg6pl r-borderRadius-z2wwpe r-borderStyle-1phboty r-borderWidth-d045u9 r-cursor-1loqt21 r-height-yq7r3z r-justifyContent-1777fci r-minHeight-j5ldn5 r-paddingHorizontal-2q4g5t r-shadow-f5u5bj r-touchAction-1otgn73"
        data-testid="ButtonTestId"
        role="button"
        tabindex="0"
      >
        <div
          class="css-text-901oao r-color-jwli3a r-fontFamily-v9d72p r-fontSize-1x35g6 r-fontWeight-b88u0q r-userSelect-lrvibr"
          dir="auto"
        >
          Test Button
        </div>
      </div>
    </div>
  </div>
  <div
    style="position: fixed; left: 0px; top: 0px; width: 0px; height: 0px; z-index: -1; overflow: hidden; visibility: hidden; transition-duration: 0.05s; transition-property: padding; transition-delay: 0s;"
  />
</body>
`;

exports[`Styled button component renders correctly renders the default primary variant correctly 1`] = `
<body>
  <div>
    <div
      class="css-view-1dbjc4n r-flex-13awgt0"
    >
      <div
        aria-label="Test Button"
        class="css-cursor-18t94o4 css-view-1dbjc4n r-alignItems-1awozwy r-alignSelf-1kihuf0 r-backgroundColor-14imykx r-borderColor-30uafw r-borderRadius-z2wwpe r-borderStyle-1phboty r-borderWidth-d045u9 r-cursor-1loqt21 r-height-yq7r3z r-justifyContent-1777fci r-minHeight-j5ldn5 r-paddingHorizontal-2q4g5t r-shadow-f5u5bj r-touchAction-1otgn73"
        data-testid="ButtonTestId"
        role="button"
        tabindex="0"
      >
        <div
          class="css-text-901oao r-color-jwli3a r-fontFamily-v9d72p r-fontSize-1x35g6 r-fontWeight-b88u0q r-userSelect-lrvibr"
          dir="auto"
        >
          Test Button
        </div>
      </div>
    </div>
  </div>
  <div
    style="position: fixed; left: 0px; top: 0px; width: 0px; height: 0px; z-index: -1; overflow: hidden; visibility: hidden; transition-duration: 0.05s; transition-property: padding; transition-delay: 0s;"
  />
</body>
`;

exports[`Styled button component renders correctly renders the default secondary variant correctly 1`] = `
<body>
  <div>
    <div
      class="css-view-1dbjc4n r-flex-13awgt0"
    >
      <div
        aria-label="Test Button"
        class="css-cursor-18t94o4 css-view-1dbjc4n r-alignItems-1awozwy r-alignSelf-1kihuf0 r-backgroundColor-14lw9ot r-borderColor-1onh70h r-borderRadius-z2wwpe r-borderStyle-1phboty r-borderWidth-d045u9 r-cursor-1loqt21 r-height-yq7r3z r-justifyContent-1777fci r-minHeight-j5ldn5 r-paddingHorizontal-2q4g5t r-shadow-f5u5bj r-touchAction-1otgn73"
        data-testid="ButtonTestId"
        role="button"
        tabindex="0"
      >
        <div
          class="css-text-901oao r-color-cq6wur r-fontFamily-v9d72p r-fontSize-1x35g6 r-fontWeight-b88u0q r-userSelect-lrvibr"
          dir="auto"
        >
          Test Button
        </div>
      </div>
    </div>
  </div>
  <div
    style="position: fixed; left: 0px; top: 0px; width: 0px; height: 0px; z-index: -1; overflow: hidden; visibility: hidden; transition-duration: 0.05s; transition-property: padding; transition-delay: 0s;"
  />
</body>
`;

exports[`Styled button component renders correctly renders the default tertiary variant correctly 1`] = `
<body>
  <div>
    <div
      class="css-view-1dbjc4n r-flex-13awgt0"
    >
      <div
        aria-label="Test Button"
        class="css-cursor-18t94o4 css-view-1dbjc4n r-alignItems-1awozwy r-alignSelf-1kihuf0 r-backgroundColor-1q56jx2 r-borderColor-1onh70h r-borderRadius-z2wwpe r-borderStyle-1phboty r-borderWidth-d045u9 r-cursor-1loqt21 r-height-yq7r3z r-justifyContent-1777fci r-minHeight-j5ldn5 r-paddingHorizontal-2q4g5t r-shadow-f5u5bj r-touchAction-1otgn73"
        data-testid="ButtonTestId"
        role="button"
        tabindex="0"
      >
        <div
          class="css-text-901oao r-color-jwli3a r-fontFamily-v9d72p r-fontSize-1x35g6 r-fontWeight-b88u0q r-userSelect-lrvibr"
          dir="auto"
        >
          Test Button
        </div>
      </div>
    </div>
  </div>
  <div
    style="position: fixed; left: 0px; top: 0px; width: 0px; height: 0px; z-index: -1; overflow: hidden; visibility: hidden; transition-duration: 0.05s; transition-property: padding; transition-delay: 0s;"
  />
</body>
`;

exports[`Styled button component renders correctly renders the slim invertedPrimary variant correctly 1`] = `
<body>
  <div>
    <div
      class="css-view-1dbjc4n r-flex-13awgt0"
    >
      <div
        aria-label="Test Button"
        class="css-cursor-18t94o4 css-view-1dbjc4n r-alignItems-1awozwy r-alignSelf-1kihuf0 r-backgroundColor-14lw9ot r-borderColor-11mg6pl r-borderRadius-z2wwpe r-borderStyle-1phboty r-borderWidth-d045u9 r-cursor-1loqt21 r-height-wy61xf r-justifyContent-1777fci r-minHeight-1el8sus r-paddingHorizontal-2q4g5t r-shadow-f5u5bj r-touchAction-1otgn73"
        data-testid="ButtonTestId"
        role="button"
        tabindex="0"
      >
        <div
          class="css-text-901oao r-color-cq6wur r-fontFamily-v9d72p r-fontSize-1x35g6 r-fontWeight-b88u0q r-userSelect-lrvibr"
          dir="auto"
        >
          Test Button
        </div>
      </div>
    </div>
  </div>
  <div
    style="position: fixed; left: 0px; top: 0px; width: 0px; height: 0px; z-index: -1; overflow: hidden; visibility: hidden; transition-duration: 0.05s; transition-property: padding; transition-delay: 0s;"
  />
</body>
`;

exports[`Styled button component renders correctly renders the slim invertedSecondary variant correctly 1`] = `
<body>
  <div>
    <div
      class="css-view-1dbjc4n r-flex-13awgt0"
    >
      <div
        aria-label="Test Button"
        class="css-cursor-18t94o4 css-view-1dbjc4n r-alignItems-1awozwy r-alignSelf-1kihuf0 r-backgroundColor-1niwhzg r-borderColor-11mg6pl r-borderRadius-z2wwpe r-borderStyle-1phboty r-borderWidth-d045u9 r-cursor-1loqt21 r-height-wy61xf r-justifyContent-1777fci r-minHeight-1el8sus r-paddingHorizontal-2q4g5t r-shadow-f5u5bj r-touchAction-1otgn73"
        data-testid="ButtonTestId"
        role="button"
        tabindex="0"
      >
        <div
          class="css-text-901oao r-color-jwli3a r-fontFamily-v9d72p r-fontSize-1x35g6 r-fontWeight-b88u0q r-userSelect-lrvibr"
          dir="auto"
        >
          Test Button
        </div>
      </div>
    </div>
  </div>
  <div
    style="position: fixed; left: 0px; top: 0px; width: 0px; height: 0px; z-index: -1; overflow: hidden; visibility: hidden; transition-duration: 0.05s; transition-property: padding; transition-delay: 0s;"
  />
</body>
`;

exports[`Styled button component renders correctly renders the slim primary variant correctly 1`] = `
<body>
  <div>
    <div
      class="css-view-1dbjc4n r-flex-13awgt0"
    >
      <div
        aria-label="Test Button"
        class="css-cursor-18t94o4 css-view-1dbjc4n r-alignItems-1awozwy r-alignSelf-1kihuf0 r-backgroundColor-14imykx r-borderColor-30uafw r-borderRadius-z2wwpe r-borderStyle-1phboty r-borderWidth-d045u9 r-cursor-1loqt21 r-height-wy61xf r-justifyContent-1777fci r-minHeight-1el8sus r-paddingHorizontal-2q4g5t r-shadow-f5u5bj r-touchAction-1otgn73"
        data-testid="ButtonTestId"
        role="button"
        tabindex="0"
      >
        <div
          class="css-text-901oao r-color-jwli3a r-fontFamily-v9d72p r-fontSize-1x35g6 r-fontWeight-b88u0q r-userSelect-lrvibr"
          dir="auto"
        >
          Test Button
        </div>
      </div>
    </div>
  </div>
  <div
    style="position: fixed; left: 0px; top: 0px; width: 0px; height: 0px; z-index: -1; overflow: hidden; visibility: hidden; transition-duration: 0.05s; transition-property: padding; transition-delay: 0s;"
  />
</body>
`;

exports[`Styled button component renders correctly renders the slim secondary variant correctly 1`] = `
<body>
  <div>
    <div
      class="css-view-1dbjc4n r-flex-13awgt0"
    >
      <div
        aria-label="Test Button"
        class="css-cursor-18t94o4 css-view-1dbjc4n r-alignItems-1awozwy r-alignSelf-1kihuf0 r-backgroundColor-14lw9ot r-borderColor-1onh70h r-borderRadius-z2wwpe r-borderStyle-1phboty r-borderWidth-d045u9 r-cursor-1loqt21 r-height-wy61xf r-justifyContent-1777fci r-minHeight-1el8sus r-paddingHorizontal-2q4g5t r-shadow-f5u5bj r-touchAction-1otgn73"
        data-testid="ButtonTestId"
        role="button"
        tabindex="0"
      >
        <div
          class="css-text-901oao r-color-cq6wur r-fontFamily-v9d72p r-fontSize-1x35g6 r-fontWeight-b88u0q r-userSelect-lrvibr"
          dir="auto"
        >
          Test Button
        </div>
      </div>
    </div>
  </div>
  <div
    style="position: fixed; left: 0px; top: 0px; width: 0px; height: 0px; z-index: -1; overflow: hidden; visibility: hidden; transition-duration: 0.05s; transition-property: padding; transition-delay: 0s;"
  />
</body>
`;

exports[`Styled button component renders correctly renders the slim tertiary variant correctly 1`] = `
<body>
  <div>
    <div
      class="css-view-1dbjc4n r-flex-13awgt0"
    >
      <div
        aria-label="Test Button"
        class="css-cursor-18t94o4 css-view-1dbjc4n r-alignItems-1awozwy r-alignSelf-1kihuf0 r-backgroundColor-1q56jx2 r-borderColor-1onh70h r-borderRadius-z2wwpe r-borderStyle-1phboty r-borderWidth-d045u9 r-cursor-1loqt21 r-height-wy61xf r-justifyContent-1777fci r-minHeight-1el8sus r-paddingHorizontal-2q4g5t r-shadow-f5u5bj r-touchAction-1otgn73"
        data-testid="ButtonTestId"
        role="button"
        tabindex="0"
      >
        <div
          class="css-text-901oao r-color-jwli3a r-fontFamily-v9d72p r-fontSize-1x35g6 r-fontWeight-b88u0q r-userSelect-lrvibr"
          dir="auto"
        >
          Test Button
        </div>
      </div>
    </div>
  </div>
  <div
    style="position: fixed; left: 0px; top: 0px; width: 0px; height: 0px; z-index: -1; overflow: hidden; visibility: hidden; transition-duration: 0.05s; transition-property: padding; transition-delay: 0s;"
  />
</body>
`;
